lib/rules/no-useless-return.js

```diff
@@ -167,6 +167,17 @@ module.exports = {
             const info = segmentInfoMap.get(segment);
 
             for (const node of info.uselessReturns) {
+                if (scopeInfo.traversedTryBlockStatements && scopeInfo.traversedTryBlockStatements.length > 0) {
+                    const blockInitialRange = scopeInfo.traversedTryBlockStatements[0].range[0];
+                    const blockFinalRange = scopeInfo.traversedTryBlockStatements[0].range[1];
+                    const returnInitialRange = node.range[0];
+                    const returnFinalRange = node.range[1];
+
+                    if (returnInitialRange >= blockInitialRange && returnFinalRange <= blockFinalRange) {
+                        continue;
+                    }
+                }
```

Comment:
I think we should check all try block statements in the list instead of just the first one. Otherwise, this would still be a false negative:

```js
function foo() {
    try {
        bar();
    }
    catch {
        try {
            baz();
            return; // false negative
        } catch {
            qux();
        }
    }
  }

```

Reply:
Also, I believe this is a new false positive:

```js
function foo() {
    try {
        bar();
        return; // false positive
    } finally {
        baz();
    }
    qux();
}

```


I'm not sure what exactly is causing it.

Reply:
>I'm not sure what exactly is causing it.

I'm looking into it. I suspect the `TryStatement:exit()` query is not working correctly if there is `finally` instead of `catch`.